vendor-extra/ContentPageBundle/src/Controller/ContentController.php

```diff
@@ -49,56 +49,16 @@ public function __invoke(Request $request, string $id) : Response
             ->then(
                 function (ResponseInterface $response) use ($request) : Response {
                     $dom = FluentDOM::load((string) $response->getBody());
-                    $dom->registerNamespace('libero', 'http://libero.pub');
-
-                    /** @var DOMNodeList|Element[] $frontList */
-                    $frontList = $dom('/libero:item/libero:front[1]');
-                    $front = $frontList->item(0);
-
-                    if (!$front instanceof Element) {
-                        throw new UnexpectedValueException('Could not find a front');
-                    }
-
-                    /** @var string $title */
-                    $title = $front('string(libero:title)');
-
-                    if ('' === $title) {
-                        throw new UnexpectedValueException('Could not find a title');
-                    }
 
                     $context = [
                         'lang' => $request->getLocale(),
                         'dir' => $this->getDirection($request->getLocale()),
                     ];
 
-                    $contentHeader = [
-                        'template' => '@LiberoPatterns/content-header.html.twig',
-                        'arguments' => [
-                            'attributes' => [],
-                            'contentTitle' => [
-                                'text' => $title,
-                            ],
-                        ],
-                    ];
-
-                    if ($context['lang'] !== $frontLang = $front->getAttribute('xml:lang')) {
-                        $contentHeader['arguments']['attributes']['lang'] = $frontLang;
-
-                        if ($context['dir'] !== $frontDir = $this->getDirection($frontLang)) {
-                            $contentHeader['arguments']['attributes']['dir'] = $frontDir;
-                        }
-                    }
-
                     return new Response(
                         $this->twig->render(
                             $this->template,
-                            array_merge(
-                                $context,
-                                [
-                                    'title' => $title,
-                                    'content' => [$contentHeader],
-                                ]
-                            )
+                            $this->contentHandler->handle($dom->documentElement, $context)
```

Comment:
Other option is not have separate controllers for each type.

Reply:
This has been updated to use a single controller, it seems